src/ast_sel_cmp.cpp

```diff
@@ -0,0 +1,550 @@
+    {
+      return *this < *w;
+    }
+    if (is_ns_eq(rhs))
+    { return name() < rhs.name(); }
+    return ns() < rhs.ns();
+  }
+
+  bool Element_Selector::operator== (const Element_Selector& rhs) const
+  {
+    return is_ns_eq(rhs) &&
+           name() == rhs.name();
+  }
+
+  bool Element_Selector::operator== (const Simple_Selector& rhs) const
+  {
+    if (Element_Selector_Ptr_Const w = Cast<Element_Selector>(&rhs))
+    {
+      return is_ns_eq(rhs) &&
+             name() == rhs.name() &&
+             *this == *w;
+    }
+    return false;
+  }
+
+  bool Pseudo_Selector::operator== (const Pseudo_Selector& rhs) const
+  {
+    if (is_ns_eq(rhs) && name() == rhs.name())
+    {
+      String_Obj lhs_ex = expression();
+      String_Obj rhs_ex = rhs.expression();
+      if (rhs_ex && lhs_ex) return *lhs_ex == *rhs_ex;
+      else return lhs_ex.ptr() == rhs_ex.ptr();
+    }
+    else return false;
+  }
+
+  bool Pseudo_Selector::operator== (const Simple_Selector& rhs) const
+  {
+    if (Pseudo_Selector_Ptr_Const w = Cast<Pseudo_Selector>(&rhs))
+    {
+      return *this == *w;
+    }
+    return is_ns_eq(rhs) &&
+           name() == rhs.name();
+  }
+
+  bool Pseudo_Selector::operator< (const Pseudo_Selector& rhs) const
+  {
+    if (is_ns_eq(rhs) && name() == rhs.name())
+    {
+      String_Obj lhs_ex = expression();
+      String_Obj rhs_ex = rhs.expression();
+      if (rhs_ex && lhs_ex) return *lhs_ex < *rhs_ex;
+      else return lhs_ex.ptr() < rhs_ex.ptr();
+    }
+    if (is_ns_eq(rhs))
+    { return name() < rhs.name(); }
+    return ns() < rhs.ns();
+  }
+
+  bool Pseudo_Selector::operator< (const Simple_Selector& rhs) const
+  {
+    if (Pseudo_Selector_Ptr_Const w = Cast<Pseudo_Selector>(&rhs))
+    {
+      return *this < *w;
+    }
+    if (is_ns_eq(rhs))
+    { return name() < rhs.name(); }
+    return ns() < rhs.ns();
+  }
+
+  bool Wrapped_Selector::operator== (const Wrapped_Selector& rhs) const
+  {
+    if (is_ns_eq(rhs) && name() == rhs.name())
+    { return *(selector()) == *(rhs.selector()); }
+    else return false;
+  }
+
+  bool Wrapped_Selector::operator== (const Simple_Selector& rhs) const
+  {
+    if (Wrapped_Selector_Ptr_Const w = Cast<Wrapped_Selector>(&rhs))
+    {
+      return *this == *w;
+    }
+    return is_ns_eq(rhs) &&
+           name() == rhs.name();
+  }
+
+  bool Wrapped_Selector::operator< (const Wrapped_Selector& rhs) const
+  {
+    if (is_ns_eq(rhs) && name() == rhs.name())
+    { return *(selector()) < *(rhs.selector()); }
+    if (is_ns_eq(rhs))
+    { return name() < rhs.name(); }
+    return ns() < rhs.ns();
+  }
+
+  bool Wrapped_Selector::operator< (const Simple_Selector& rhs) const
+  {
+    if (Wrapped_Selector_Ptr_Const w = Cast<Wrapped_Selector>(&rhs))
+    {
+      return *this < *w;
+    }
+    if (is_ns_eq(rhs))
+    { return name() < rhs.name(); }
+    return ns() < rhs.ns();
+  }
+
+  bool Compound_Selector::operator== (const Compound_Selector& rhs) const
```

Comment:
Looks like this reverted https://github.com/sass/libsass/commit/2a4935c9ad137bedb1be6690a78995a6714a7e62

I wonder if there are others

Reply:
It might be possible that due to the rebase it will come in a later PR (I remember that I manually re-added this change to some commit). But thx for checking, I will check myself again

Reply:
Those previous changes should be part of this PR now too!